use-cases/rag-on-gke/alloyDB/init-database-objects/init_database.tf

```diff
@@ -0,0 +1,204 @@
+  depends_on = [
+    kubernetes_service_account.dba_service_account
+  ]
+}
+
+module "create_extension_pgvector" {
+  source               = "./kube-psql-job"
+  project_id           = var.project_id
+  name                 = "init-extension-pgvector"
+  gke_cluster_name     = var.gke_cluster_name
+  gke_cluster_location = var.gke_cluster_location
+  sql_script           = <<-EOT
+  CREATE EXTENSION IF NOT EXISTS vector;
+EOT
+  environs             = {}
+  pghost               = data.external.alloydb_primary_instance_ip.result.ipAddress
+  pgdatabase           = "ragdb"
+  k8s_namespace        = var.k8s_namespace
+  k8s_service_account  = var.dba_service_account
+  depends_on = [
+    kubernetes_service_account.dba_service_account,
+    module.createdb
+  ]
+}
+
+module "create_extension_ml_integration" {
+  count = 0 if var.skip_ml_integration else 1
+  source               = "./kube-psql-job"
+  project_id           = var.project_id
+  name                 = "init-extension-ml-integration"
+  gke_cluster_name     = var.gke_cluster_name
+  gke_cluster_location = var.gke_cluster_location
+  sql_script           = <<-EOT
+  CREATE EXTENSION IF NOT EXISTS google_ml_integration VERSION '1.3';
+  GRANT SELECT, INSERT, UPDATE, DELETE
+    ON ALL TABLES IN SCHEMA google_ml
+    TO "alloydb-raguser@${var.project_id}.iam";
+  GRANT EXECUTE
+    ON ALL FUNCTIONS IN SCHEMA google_ml
+    TO "alloydb-raguser@${var.project_id}.iam";
+  GRANT ALL
+    ON SCHEMA google_ml
+    TO "alloydb-raguser@${var.project_id}.iam";
+  GRANT ALL
+    ON ALL TABLES IN SCHEMA public
+    TO "alloydb-raguser@${var.project_id}.iam";
+  GRANT ALL
+    ON ALL FUNCTIONS IN SCHEMA public
+    TO "alloydb-raguser@${var.project_id}.iam";
+  GRANT ALL ON SCHEMA public
+    TO "alloydb-raguser@${var.project_id}.iam";
+EOT
+  environs             = {}
+  pghost               = data.external.alloydb_primary_instance_ip.result.ipAddress
+  pgdatabase           = "ragdb"
+  k8s_namespace        = var.k8s_namespace
+  k8s_service_account  = var.dba_service_account
+  depends_on = [
+    kubernetes_service_account.dba_service_account,
+    module.createdb
+  ]
+}
+
+module "create_in_db_objects" {
+  count = 0 if var.skip_ml_integration else 1
+  source               = "./kube-psql-job"
+  project_id           = var.project_id
+  name                 = "create-in-db-objs"
+  gke_cluster_name     = var.gke_cluster_name
+  gke_cluster_location = var.gke_cluster_location
+  sql_script           = file("${path.module}/assets/ml-integration.sql")
+  environs = {
+    "PRETRAINED_MODEL_EP" = var.pretrained_model_endpoint
+    "EMBEDDING_ENDPOINT"  = var.embedding_endpoint
+  }
+  pghost              = data.external.alloydb_primary_instance_ip.result.ipAddress
+  pgdatabase          = "ragdb"
+  k8s_namespace       = var.k8s_namespace
+  k8s_service_account = var.rag_service_account
+  depends_on = [
+    kubernetes_service_account.rag_service_account,
+    module.create_extension
+  ]
+}
```

Comment:
All of this should be moved out to a Kubernetes job to initialize the database as part of the application. Terraform should only be used to stand up the initial infrastructure and cluster level configuration.